src/sample.py

```diff
@@ -39,7 +40,7 @@ def message (self, buf, font, msg, data, _x_what_is_this):
 		return True
 debugger = Debugger()
 hb.buffer_set_message_func (buf, debugger.message, 1, 0)
-hb.buffer_add_utf8 (buf, text.encode('utf-8'), 0, -1)
+hb.buffer_add_utf32 (buf, codepoints, 0, -1)
```

Comment:
and do you need len(codepoints) for the add_codepoints() function too?

On Sat, Jun 18, 2016 at 11:22 AM, Kelvin Ma <kelvinsthirteen@gmail.com>
wrote:

> so that’s what the -1 does…what does the 0 do then?
>
> On Sat, Jun 18, 2016 at 10:25 AM, Khaled Hosny <notifications@github.com>
> wrote:
>
>> In src/sample.py
>> <https://github.com/behdad/harfbuzz/pull/269#discussion_r67601692>:
>>
>> > @@ -39,7 +40,7 @@ def message (self, buf, font, msg, data, _x_what_is_this):
>> >  		return True
>> >  debugger = Debugger()
>> >  hb.buffer_set_message_func (buf, debugger.message, 1, 0)
>> > -hb.buffer_add_utf8 (buf, text.encode('utf-8'), 0, -1)
>> > +hb.buffer_add_utf32 (buf, codepoints, 0, -1)
>>
>> -1 should be len(codepoints) since your list is not NULL-terminated.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/behdad/harfbuzz/pull/269/files/88f9f6da65ce5187bb43ebd0cf470955bf8f84da#r67601692>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/ACcEOi__X9y4lNDo4oAxizYQgVL-PANoks5qM__QgaJpZM4I3_8R>
>> .
>>
>
>